### ISSUE TITLE
Installing Class 90/Mk3 DVT pack downloads and installs wrong branding patch/Freightliner 90 pack doesn't download BP.

### ISSUE THREAD

**Issue by Slenderman7676** [shipsaw/AP-Downloader#2]
This is a very useful tool except tonight I bought both the Class 90/Mk3 DVT pack and the Class 90 Freightliner pack and installed them. It downloaded and installed fine except I carefully was reading the prompt during the installs and as well as looking in the folder where all files are downloaded to. I noticed two things.
The installer downloaed and installed the branding patch for the Class 350 Enhancement pack instead of the correct branding patch for the 90 packs. And the installer didn't even download/install the branding patch for the Freightliner 90 pack. I assume this is an issue with the product database. I can't find any workarounds accept for having to manually download and install the correct patches from the AP website.

**Comment by shipsaw** [comment]
@Slenderman7676 thanks, discovered that the last DB Update had several incorrect values, I have updated the endpoint that the app uses to update the product db, after restarting AP Downloader this issue should be corrected

**Comment by Slenderman7676** [comment]
Thanks. On an unrelated note. The installer doesn't seem to even know that Wherry Lines 2.0 exists. Like it is not listed at all.

**Comment by shipsaw** [comment]
I'll look into the wherry lines one

**Comment by shipsaw** [comment]
@Slenderman7676 do you have the Standalone Wherry Lines 2.0?  The issue is I only have the upgrade, so don't have that info to populate the database.  To add it, I would need the download link, the filename when you download it, and the image link on your downloads page.  I can't actually access any of these links as I have not bought that version, but I can make it so it works for you.

**Comment by Slenderman7676** [comment]
I have got the standalone 2.0.

**Comment by shipsaw** [comment]
So because I can't get this pack as I have the Upgrade, I need the following info to add to the DB:
1) To get the download link, I need the product ID.  When you go to your downloads, right click on a download link and copy-paste that to notepad etc... you should get something like: `https://www.armstrongpowerhouse.com/index.php?route=account/download/download&download_id=214`, the number I need in this example is 214
2) I need the name of the downloaded file, ex `AP_Mk3A-B_Coach_Pack.zip`
3) The link for the image in the downloads that shows the product `https://www.armstrongpowerhouse.com/image/cache/catalog/APMK3A-B_Cover-238x294.png`

**Comment by Slenderman7676** [comment]
Okay. This should be of help.

211

AP_Wherry_Lines_Route_Pack_2.0.zip

https://www.armstrongpowerhouse.com/image/cache/catalog/APWLV2_Cover-238x294.png

**Comment by shipsaw** [comment]
@Slenderman7676 updated the DB, does the route show up for you now?

**Comment by Slenderman7676** [comment]
@shipsaw  Yes, it does. Thank you. In future perhaps it would be feasable to have a product DB megathread where users can describe current issues regarding the products DB. Like I've got another one here.

When installing either of the Class 37 Loco packs, the Branding patch is not downloaded or installed and is required to be installed manually.

Product IDs:
37 V1: 184
37 V2: 198

Relevant branding pack links:
37 V1: https://armstrongpowerhouse.com/free_download/Patches/AP_Class_37_Locomotive_Pack_Vol_1_Branding_Patch.zip
37 V2: https://armstrongpowerhouse.com/free_download/Patches/AP_Class_37_Locomotive_Pack_Vol_2_Branding_Patch.zip

**Comment by shipsaw** [comment]
Alright Updated the productsDB with those class 37 issues.

I'll look into the thread idea, another thing I'm trying to set up at the moment is automatic CSV generation on build or commit, so that diagnosing these DB issues will be easier for people in the future.

Thanks again for all the assistance on finding those problems

**Comment by shipsaw** [comment]
So I've added some hooks to generate a CSV file every time I do a commit, if you look at [ApProducts.csv](https://github.com/shipsaw/AP-Downloader/blob/master/ApProducts.csv) it will open a table with all the entries in the DB.  This should make it easier to diagnose / correct issues
